src/atomicvar.h

```diff
@@ -124,12 +138,23 @@
 } while(0)
 #define atomicSetWithSync(var,value) \
     __atomic_store_n(&var,value,__ATOMIC_SEQ_CST)
+#define atomicFlagTestSet(var, testres_var) \
+    testres_var = __atomic_test_and_set(&var,memory_order_relaxed)
+#define atomicFlagClear(var) \
+    __atomic_clear(&var,memory_order_relaxed)
 #define REDIS_ATOMIC_API "atomic-builtin"
 
 #elif defined(HAVE_ATOMIC)
 /* Implementation using __sync macros. */
 
+/* According to Intel Itanium Processor-specific Application Binary Interface, section 7.4, 
+__sync operations on booleans expect int operand. */
+#undef redisAtomicFlag
+#define redisAtomicFlag int
```

Comment:
we cannot make intel specific assumptions here.
at least not without making this code use an alternative sync primitive, or at the very least break the compilation when that's not the case.

Reply:
according to [6.54 Legacy __sync Built-in Functions for Atomic Memory Access](https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html)

> The following built-in functions are intended to be compatible with those described in the Intel Itanium Processor-specific Application Binary Interface, section 7.4

Reply:
ok, maybe i misunderstood.
this comment applies for the entire block that uses the `_sync` intrinsics, not just the ones you added?
i.e. these intrinsics will be supported by the compiler only when using a supported platform, and if not, we'll either choose a different atomic option, or fail to build.

so your comment is just meant to improve the description in the code, and it doesn't mean that the changes you made could perform wrongly when not used on Intel?

Reply:
> this comment applies for the entire block that uses the _sync intrinsics, not just the ones you added?
It is actually, but I added this comment since there was a misalignment between the GCC doc and the intel guide regarding the return type of `__sync_bool_compare_and_swap` (bool in GCC, int in intel)
Also, intel describes which types of operands the function expects (int, long, long long, unsigned int, unsigned long, unsigned long long)

> so your comment is just meant to improve the description in the code, and it doesn't mean that the changes you made could perform wrongly when not used on Intel?

So yes, only to explain.